File: 5/5.30.dart
class Equipment {}

class Equipments {
  bool get canChange => true; //本来はロジックが入ってtrue/falseが返る
  Equipment head = Equipment();
  Equipment armor = Equipment();
  Equipment arm = Equipment();
}

class Member {
  final Equipments equipments = Equipments();
}

class Party {
  final List<Member> members = [Member(), Member(), Member()];
}

void equipArmor(Party party, int memberId, Equipment newAromor) {
  // [BAD]
  // パラメータの中のパラメータの中のパラメータ...とどんどん深いところのパラメータにアクセスしている
  // カプセル化できておらず、これではどこからでもアクセス可能であり、どこでどうarmorが変わったのか分からなくなる
  if (party.members[memberId].equipments.canChange) {
    party.members[memberId].equipments.armor = newAromor;
  }
}

File: 5/5.30.swift
class Equipment {}

class Equipments {
    var canChange: Bool {
        true //本来はロジックが入ってtrue/falseが返る
    }
    var head: Equipment = .init()
    var armor: Equipment = .init()
    var arm: Equipment = .init()
}

class Member {
    let equipments: Equipments = .init()
}

class Party {
    let members: [Member] = [Member(), Member(), Member()]
}

func equipArmor(party: Party, memberId: Int, newAromor: Equipment) {
    // [BAD]
    // パラメータの中のパラメータの中のパラメータ...とどんどん深いところのパラメータにアクセスしている
    // カプセル化できておらず、これではどこからでもアクセス可能であり、どこでどうarmorが変わったのか分からなくなる
    guard party.members[memberId].equipments.canChange else {
        return
    }
    party.members[memberId].equipments.armor = newAromor
}
